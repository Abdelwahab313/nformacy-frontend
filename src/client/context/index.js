import React, { createContext, useContext, useReducer } from 'react';
import {
  SET_ERROR_MESSAGE,
  UPDATE_CLIENTS,
  UPDATE_CURRENT_CLIENT,
  VERIFY_CLIENT,
  VERIFY_CURRENT_CLIENT,
} from './actionTypes';

const ClientContext = createContext();
const defaultValues = {
  clients: [],
  errorMessage: '',
};

const ClientProvider = ({ children, initialValue }) => {
  const { Provider } = ClientContext;
  initialValue = { ...defaultValues, ...initialValue };
  return (
    <Provider value={useReducer(clientReducer, initialValue)}>
      {children}
    </Provider>
  );
};

const useClientState = () => {
  const context = useContext(ClientContext);
  if (context === undefined) {
    throw new Error('useClientState must be used within a ClientProvider');
  }
  return context;
};

const clientReducer = (state, action) => {
  switch (action.type) {
    case UPDATE_CLIENTS:
      return {
        ...state,
        clients: action.payload,
      };
    case UPDATE_CURRENT_CLIENT:
      return {
        ...state,
        currentClient: action.payload,
      };
    case VERIFY_CURRENT_CLIENT:
      const currentClient = state.currentClient;
      currentClient.verified = true;
      return {
        ...state,
        currentClient: currentClient,
      };
    case VERIFY_CLIENT:
      const newClients = state.clients.map((client) => {
        if (client[0] === action.payload.uuid) {
          client[6] = true;
          return client;
        } else {
          return client;
        }
      });
      return {
        ...state,
        clients: newClients,
      };
    case SET_ERROR_MESSAGE:
      return {
        ...state,
        errorMessage: action.payload,
      };
  }
};

export { ClientProvider, useClientState };
