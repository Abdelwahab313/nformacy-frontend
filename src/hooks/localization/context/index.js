import React, { createContext, useContext, useReducer } from 'react';

const LocaleContext = createContext();
const DEFAULT_LOCALE = 'en';
const LOCALES = {
  en: 'en',
  ar: 'ar',
};
export const CHANGE_LOCALE = 'CHANGE_LOCALE';

export const useLocaleContext = () => {
  const context = useContext(LocaleContext);
  if (context === undefined)
    throw new Error('useLocale should only be used within locale provider');
  return context;
};

export const LocaleProvider = ({
  children,
  initialLocale = DEFAULT_LOCALE,
}) => {
  const { Provider } = LocaleContext;
  return (
    <Provider value={useReducer(localeReducer, initialLocale)}>{children}</Provider>
  );
};

const localeReducer = (state, action) => {
  if (action.type === CHANGE_LOCALE) {
    return action.payload.locale;
  }
  return LOCALES.en;
};
