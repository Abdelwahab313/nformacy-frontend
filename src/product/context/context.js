import React, { createContext, useContext, useReducer } from 'react';
import {
  CLOSE_INSERT_DIALOG_AND_SAVE,
  CLOSE_UPDATE_DIALOG_AND_UPDATE,
  CLOSE_INSERT_DIALOG_WITHOUT_SAVE,
  CLOSE_UPDATE_DIALOG_WITHOUT_SAVE,
  OPEN_INSERT_DIALOG,
  OPEN_UPDATE_DIALOG,
  SET_ERROR,
  UPDATE_PRODUCTS,
} from './contextActions';

const ProductContext = createContext();
const initValues = {
  products: [],
  insertDialogOpened: false,
  updateDialogOpened: false,
  errorMessage: '',
};

const ProductProvider = ({ children, initialValue }) => {
  initialValue = { ...initValues, ...initialValue };
  const { Provider } = ProductContext;
  return (
    <Provider value={useReducer(productReducer, initialValue)}>
      {children}
    </Provider>
  );
};

const useProductState = () => {
  const context = useContext(ProductContext);
  if (context === undefined) {
    throw new Error('UseProductState must be used within a ProductProvider');
  }
  return context;
};

const productReducer = (state, action) => {
  switch (action.type) {
    case UPDATE_PRODUCTS:
      return { ...state, products: action.payload };
    case OPEN_INSERT_DIALOG:
      return { ...state, insertDialogOpened: true };
    case OPEN_UPDATE_DIALOG:
      return { ...state, updateDialogOpened: true };
    case CLOSE_INSERT_DIALOG_AND_SAVE:
      return {
        ...state,
        insertDialogOpened: false,
        products: [...state.products, action.payload],
      };
    case CLOSE_UPDATE_DIALOG_AND_UPDATE:
      let toBeEditedIndex = state.products.findIndex(
        (product) => product[0] === action.payload[0],
      );
      state.products[toBeEditedIndex] = action.payload;
      return {
        ...state,
        updateDialogOpened: false,
        products: state.products,
      };
    case CLOSE_INSERT_DIALOG_WITHOUT_SAVE:
      return {
        ...state,
        insertDialogOpened: false,
      };
    case CLOSE_UPDATE_DIALOG_WITHOUT_SAVE:
      return {
        ...state,
        updateDialogOpened: false,
      };
    case SET_ERROR:
      return {
        ...state,
        errorMessage: action.payload,
      };
    default:
      throw new Error(`Unhandled action type: ${action.type}`);
  }
};

export { useProductState, ProductProvider };
