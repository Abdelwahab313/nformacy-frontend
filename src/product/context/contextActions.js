const CLOSE_INSERT_DIALOG_AND_SAVE = 'CLOSE_DIALOG_AND_SAVE';
const CLOSE_INSERT_DIALOG_WITHOUT_SAVE = 'CLOSE_DIALOG_WITHOUT_SAVE';
const CLOSE_UPDATE_DIALOG_AND_UPDATE = 'CLOSE_DIALOG_AND_UPDATE';
const SET_ERROR = 'SET_ERROR';
const UPDATE_PRODUCTS = 'UPDATE_PRODUCTS';
const OPEN_INSERT_DIALOG = 'OPEN_DIALOG';
const OPEN_UPDATE_DIALOG = 'OPEN_UPDATE_DIALOG';
const CLOSE_UPDATE_DIALOG_WITHOUT_SAVE = 'CLOSE_UPDATE_DIALOG_WITHOUT_SAVE';

export {
  CLOSE_INSERT_DIALOG_AND_SAVE,
  SET_ERROR,
  CLOSE_INSERT_DIALOG_WITHOUT_SAVE,
  CLOSE_UPDATE_DIALOG_AND_UPDATE,
  OPEN_INSERT_DIALOG,
  OPEN_UPDATE_DIALOG,
  UPDATE_PRODUCTS,
  CLOSE_UPDATE_DIALOG_WITHOUT_SAVE,
};
