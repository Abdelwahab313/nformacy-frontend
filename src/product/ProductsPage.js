import React, { useRef, useState } from 'react';
import Grid from '@material-ui/core/Grid';
import {
  AppBar,
  Button,
  Dialog,
  IconButton,
  Slide,
  Toolbar,
  Typography,
} from '@material-ui/core';
import MUIDataTable from 'mui-datatables';
import CircularProgress from '@material-ui/core/CircularProgress';
import CloseIcon from '@material-ui/icons/Close';
import ErrorDialog from '../components/errors/ErrorDialog';
import AddProductPage from './AddProductPage';
import productsStyle from './styles/productsScreen';
import { useProductState } from './context/context';
import {
  CLOSE_INSERT_DIALOG_WITHOUT_SAVE,
  CLOSE_UPDATE_DIALOG_WITHOUT_SAVE,
  OPEN_INSERT_DIALOG,
  OPEN_UPDATE_DIALOG,
  SET_ERROR,
} from './context/contextActions';
import TableSelectedToolBar from './components/TableSelectedToolBar';
import useProductFetcher from './hooks/ProductFetcher';
import EditProductPage from './EditProductPage';
import { tableLabels } from '../constants/tableLocalization';

const Transition = React.forwardRef(function Transition(props, ref) {
  return <Slide direction='up' ref={ref} {...props} />;
});

const ProductsPage = () => {
  const classes = productsStyle();
  const [
    { products, insertDialogOpened, updateDialogOpened, errorMessage },
    dispatch,
  ] = useProductState();
  const { productsLoading } = useProductFetcher();
  let selectedProduct = useRef(undefined);

  if (productsLoading) {
    return (
      <div className={classes.progressContainer}>
        <CircularProgress />
      </div>
    );
  } else {
    return (
      <div className={classes.root} dir='rtl'>
        {errorMessage && (
          <ErrorDialog
            message={errorMessage}
            close={() => {
              dispatch({ type: SET_ERROR, payload: '' });
              dispatch({
                type: SET_ERROR,
                payload: '',
              });
            }}
          />
        )}
        <Grid className={classes.tableContainer}>
          <Button
            className={classes.addButton}
            variant='contained'
            id={'add-product-button'}
            onClick={() =>
              dispatch({ type: OPEN_INSERT_DIALOG, payload: true })
            }
            color='primary'>
            اضافه منتج جديد
          </Button>
          <MUIDataTable
            title={'البضائع'}
            data={products}
            columns={[
              {
                name: 'id',
                options: {
                  filter: false,
                  display: 'excluded',
                },
              },
              {
                name: 'اسم المنتج',
                options: {
                  filter: false,
                },
              },
              {
                name: 'SKU',
                options: {
                  filter: false,
                },
              },
              {
                name: 'سعر المنتج',
                options: {
                  filter: false,
                },
              },
            ]}
            options={{
              responsive: 'scrollMaxHeight',
              selectableRowsOnClick: true,
              selectableRows: 'single',
              filter: false,
              rowsPerPageOptions: [5, 10, 20],
              customToolbarSelect: (
                selectedRows,
                displayData,
                setSelectedRows,
              ) => {
                const selectedProductIndex = selectedRows.data[0].dataIndex;
                selectedProduct.current = products[selectedProductIndex];
                return (
                  <TableSelectedToolBar
                    onUpdate={() => dispatch({ type: OPEN_UPDATE_DIALOG })}
                  />
                );
              },
              ...tableLabels,
            }}
          />
        </Grid>
        <Dialog
          fullScreen
          open={insertDialogOpened}
          TransitionComponent={Transition}
          id={'add-user-dialog'}>
          <AppBar>
            <Toolbar className={classes.toolBar}>
              <IconButton
                edge='start'
                id={'close_insert_product_btn'}
                onClick={() =>
                  dispatch({
                    type: CLOSE_INSERT_DIALOG_WITHOUT_SAVE,
                  })
                }
                aria-label='close'>
                <CloseIcon />
              </IconButton>
              <Typography>اضافه منتج جديد</Typography>
            </Toolbar>
          </AppBar>
          <AddProductPage />
        </Dialog>
        <Dialog
          fullScreen
          open={updateDialogOpened}
          TransitionComponent={Transition}
          id={'add-user-dialog'}>
          <AppBar>
            <Toolbar className={classes.toolBar}>
              <IconButton
                edge='start'
                onClick={() =>
                  dispatch({
                    type: CLOSE_UPDATE_DIALOG_WITHOUT_SAVE,
                  })
                }
                aria-label='close'>
                <CloseIcon />
              </IconButton>
              <Typography>تعديل منتج</Typography>
            </Toolbar>
          </AppBar>
          <EditProductPage product={selectedProduct.current} />
        </Dialog>
      </div>
    );
  }
};

export default ProductsPage;
