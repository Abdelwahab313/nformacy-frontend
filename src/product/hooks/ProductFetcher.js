import React, { useEffect, useState } from 'react';
import { useProductState } from '../context/context';
import { useAuth } from '../../context/auth';
import { SET_ERROR, UPDATE_PRODUCTS } from '../context/contextActions';
import { fetchProducts } from '../productsApi';
import { convertObjToArray } from '../utils';

const ProductFetcher = () => {
  const [_, dispatch] = useProductState();
  const { authTokens, setAuthTokens } = useAuth();
  const [productsLoading, setProductsLoading] = useState(false);

  function sortByDate(data) {
    const fetchedProducts = data;
    fetchedProducts.sort((a, b) => new Date(a.created) - new Date(b.created));
    return fetchedProducts;
  }

  function handleApiErrors(reason) {
    if (reason.message === 'Network Error') {
      dispatch({
        type: SET_ERROR,
        payload: 'حدث خطأ أثناء الاتصال بالخادم',
      });
    } else if (reason.response.status === 401) {
      localStorage.removeItem('tokens');
      localStorage.removeItem('users');
      setAuthTokens();
    }
  }

  function adaptDataForPresentation(sortedProducts) {
    return sortedProducts.map((product) => convertObjToArray(product));
  }

  function getProducts() {
    setProductsLoading(true);
    return fetchProducts(authTokens)
      .then((res) => {
        return sortByDate(res.data);
      })
      .then((sortedProducts) => {
        const adaptedProducts = adaptDataForPresentation(sortedProducts);
        dispatch({ type: UPDATE_PRODUCTS, payload: adaptedProducts });
      })
      .catch((reason) => {
        handleApiErrors(reason);
      });
  }

  useEffect(() => {
    getProducts().finally(() => {
      setProductsLoading(false);
    });
  }, [setProductsLoading]);
  return { productsLoading };
};
export default ProductFetcher;
