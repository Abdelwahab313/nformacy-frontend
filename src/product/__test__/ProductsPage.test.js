import * as AuthContextMoudle from '../../context/auth';
import { AuthContext } from '../../context/auth';
import React from 'react';
import ProductsPage from '../ProductsPage';
import { ProductProvider } from '../context/context';
import { render } from '@testing-library/react';
import ProductFetcher from '../hooks/ProductFetcher';
jest.mock('../hooks/ProductFetcher');
describe('ProductFetcher Presentation', () => {
  it('should render products list screen correctly', () => {
    jest.spyOn(AuthContextMoudle, 'useAuth').mockImplementation(() => ({
      authTokens: { access_token: 'testToken' },
    }));
    ProductFetcher.mockImplementationOnce(() => [false, 'setter']);
    const products = [
      {
        uuid: 'test_uuid1',
        name: 'test_product1',
        sku: '1',
        price: 50,
      },
      {
        uuid: 'test_uuid2',
        name: 'test_product2',
        sku: '2',
        price: 60,
      },
    ];
    const { asFragment } = render(
      <AuthContext.Provider>
        <ProductProvider initialValue={{ products: products }}>
          <ProductsPage />
        </ProductProvider>
      </AuthContext.Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
