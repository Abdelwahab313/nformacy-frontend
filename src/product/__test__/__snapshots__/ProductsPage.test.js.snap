// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductFetcher Presentation should render products list screen correctly 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
    dir="rtl"
  >
    <div
      class="MuiGrid-root makeStyles-tableContainer-5"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-addButton-2 MuiButton-containedPrimary"
        id="add-product-button"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          اضافه منتج جديد
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiPaper-root MuiPaper-elevation4 MUIDataTable-paper-142 MuiPaper-rounded"
      >
        <div
          aria-label="Table Toolbar"
          class="MuiToolbar-root MuiToolbar-regular MUIDataTableToolbar-root-177 MuiToolbar-gutters"
          role="toolbar"
        >
          <div
            class="MUIDataTableToolbar-left-178"
          >
            <div
              aria-hidden="true"
              class="MUIDataTableToolbar-titleRoot-180"
            >
              <h6
                class="MuiTypography-root MUIDataTableToolbar-titleText-181 MuiTypography-h6"
              >
                البضائع
              </h6>
            </div>
          </div>
          <div
            class="MUIDataTableToolbar-actions-179"
          >
            <button
              aria-label="بحث"
              class="MuiButtonBase-root MuiIconButton-root MUIDataTableToolbar-icon-182"
              data-testid="بحث-iconButton"
              tabindex="0"
              title="بحث"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="CSV تحميل"
              class="MuiButtonBase-root MuiIconButton-root MUIDataTableToolbar-icon-182"
              data-testid="CSV تحميل-iconButton"
              tabindex="0"
              title="CSV تحميل"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM17 13l-5 5-5-5h3V9h4v4h3z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <span>
              <span>
                <button
                  aria-label="طباعه"
                  class="MuiButtonBase-root MuiIconButton-root MUIDataTableToolbar-icon-182"
                  data-testid="طباعه-iconButton"
                  tabindex="0"
                  title="طباعه"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 8H5c-1.66 0-3 1.34-3 3v6h4v4h12v-4h4v-6c0-1.66-1.34-3-3-3zm-3 11H8v-5h8v5zm3-7c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1zm-1-9H6v4h12V3z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </span>
            </span>
            <span>
              <button
                aria-label="عرض"
                class="MuiButtonBase-root MuiIconButton-root MUIDataTableToolbar-icon-182"
                data-testid="عرض-iconButton"
                tabindex="0"
                title="عرض"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 18h5V5h-5v13zm-6 0h5V5H4v13zM16 5v13h5V5h-5z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </span>
          </div>
        </div>
        <div
          class="MUIDataTableFilterList-root-249"
        />
        <div
          class="MUIDataTable-responsiveScrollMaxHeight-145"
          style="position: relative;"
        >
          <table
            class="MuiTable-root MUIDataTable-tableRoot-143"
            role="grid"
            tabindex="0"
          >
            <caption
              class="MUIDataTable-caption-148"
            >
              البضائع
            </caption>
            <thead
              class="MuiTableHead-root MUIDataTableHead-main-253"
            >
              <tr
                class="MuiTableRow-root MUIDataTableHeadRow-root-256 MuiTableRow-head"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MUIDataTableSelectCell-root-262 MUIDataTableSelectCell-fixedHeaderCommon-264 MUIDataTableSelectCell-fixedHeaderXAxis-265 MUIDataTableSelectCell-fixedHeaderYAxis-266 MUIDataTableSelectCell-headerCell-270 MuiTableCell-paddingCheckbox"
                  scope="col"
                >
                  <div
                    style="display: flex; align-items: center;"
                  />
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MUIDataTableHeadCell-root-287 MUIDataTableHeadCell-fixedHeaderCommon-289 MUIDataTableHeadCell-fixedHeaderXAxis-290 MUIDataTableHeadCell-fixedHeaderYAxis-291"
                  scope="col"
                >
                  <span
                    class="MUIDataTableHeadCell-toolButton-298"
                    role="button"
                    tabindex="0"
                    title="ترتيب حسب اسم المنتج"
                  >
                    <div
                      class="MUIDataTableHeadCell-data-294"
                    >
                      اسم المنتج
                    </div>
                    <div
                      class="MUIDataTableHeadCell-sortAction-295"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiTableSortLabel-root MUIDataTableHeadCell-sortLabelRoot-296"
                        role="button"
                        tabindex="0"
                      />
                    </div>
                  </span>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MUIDataTableHeadCell-root-287 MUIDataTableHeadCell-fixedHeaderCommon-289 MUIDataTableHeadCell-fixedHeaderXAxis-290 MUIDataTableHeadCell-fixedHeaderYAxis-291"
                  scope="col"
                >
                  <span
                    class="MUIDataTableHeadCell-toolButton-298"
                    role="button"
                    tabindex="0"
                    title="ترتيب حسب SKU"
                  >
                    <div
                      class="MUIDataTableHeadCell-data-294"
                    >
                      SKU
                    </div>
                    <div
                      class="MUIDataTableHeadCell-sortAction-295"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiTableSortLabel-root MUIDataTableHeadCell-sortLabelRoot-296"
                        role="button"
                        tabindex="0"
                      />
                    </div>
                  </span>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MUIDataTableHeadCell-root-287 MUIDataTableHeadCell-fixedHeaderCommon-289 MUIDataTableHeadCell-fixedHeaderXAxis-290 MUIDataTableHeadCell-fixedHeaderYAxis-291"
                  scope="col"
                >
                  <span
                    class="MUIDataTableHeadCell-toolButton-298"
                    role="button"
                    tabindex="0"
                    title="ترتيب حسب سعر المنتج"
                  >
                    <div
                      class="MUIDataTableHeadCell-data-294"
                    >
                      سعر المنتج
                    </div>
                    <div
                      class="MUIDataTableHeadCell-sortAction-295"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiTableSortLabel-root MUIDataTableHeadCell-sortLabelRoot-296"
                        role="button"
                        tabindex="0"
                      />
                    </div>
                  </span>
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root"
            >
              <tr
                class="MuiTableRow-root MUIDataTableBodyRow-root-309 MUIDataTableBodyRow-hover-310 MUIDataTableBodyRow-hoverCursor-311 MuiTableRow-hover"
                data-testid="MUIDataTableBodyRow-0"
                id="MUIDataTableBodyRow-0"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MUIDataTableSelectCell-root-262 MUIDataTableSelectCell-fixedHeaderCommon-264 MUIDataTableSelectCell-fixedHeaderXAxis-265 MUIDataTableSelectCell-fixedHeaderYAxis-266 MuiTableCell-paddingCheckbox"
                >
                  <div
                    style="display: flex; align-items: center;"
                  >
                    <span
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-319 MuiCheckbox-root MUIDataTableSelectCell-checkboxRoot-272 MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <input
                          class="PrivateSwitchBase-input-322"
                          data-indeterminate="false"
                          id="MUIDataTableSelectCell-0"
                          type="checkbox"
                          value=""
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </span>
                  </div>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MUIDataTableBodyCell-root-323 MUIDataTableBodyCell-cellHide-324"
                >
                  اسم المنتج
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MUIDataTableBodyCell-root-323"
                  data-testid="MuiDataTableBodyCell-1-0"
                />
                <td
                  class="MuiTableCell-root MuiTableCell-body MUIDataTableBodyCell-root-323 MUIDataTableBodyCell-cellHide-324"
                >
                  SKU
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MUIDataTableBodyCell-root-323"
                  data-testid="MuiDataTableBodyCell-2-0"
                />
                <td
                  class="MuiTableCell-root MuiTableCell-body MUIDataTableBodyCell-root-323 MUIDataTableBodyCell-cellHide-324"
                >
                  سعر المنتج
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MUIDataTableBodyCell-root-323"
                  data-testid="MuiDataTableBodyCell-3-0"
                />
              </tr>
              <tr
                class="MuiTableRow-root MUIDataTableBodyRow-root-309 MUIDataTableBodyRow-hover-310 MUIDataTableBodyRow-hoverCursor-311 MuiTableRow-hover"
                data-testid="MUIDataTableBodyRow-1"
                id="MUIDataTableBodyRow-1"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MUIDataTableSelectCell-root-262 MUIDataTableSelectCell-fixedHeaderCommon-264 MUIDataTableSelectCell-fixedHeaderXAxis-265 MUIDataTableSelectCell-fixedHeaderYAxis-266 MuiTableCell-paddingCheckbox"
                >
                  <div
                    style="display: flex; align-items: center;"
                  >
                    <span
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-319 MuiCheckbox-root MUIDataTableSelectCell-checkboxRoot-272 MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <input
                          class="PrivateSwitchBase-input-322"
                          data-indeterminate="false"
                          id="MUIDataTableSelectCell-1"
                          type="checkbox"
                          value=""
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </span>
                  </div>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MUIDataTableBodyCell-root-323 MUIDataTableBodyCell-cellHide-324"
                >
                  اسم المنتج
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MUIDataTableBodyCell-root-323"
                  data-testid="MuiDataTableBodyCell-1-1"
                />
                <td
                  class="MuiTableCell-root MuiTableCell-body MUIDataTableBodyCell-root-323 MUIDataTableBodyCell-cellHide-324"
                >
                  SKU
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MUIDataTableBodyCell-root-323"
                  data-testid="MuiDataTableBodyCell-2-1"
                />
                <td
                  class="MuiTableCell-root MuiTableCell-body MUIDataTableBodyCell-root-323 MUIDataTableBodyCell-cellHide-324"
                >
                  سعر المنتج
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MUIDataTableBodyCell-root-323"
                  data-testid="MuiDataTableBodyCell-3-1"
                />
              </tr>
            </tbody>
          </table>
        </div>
        <table
          class="MuiTable-root"
        >
          <tfoot
            class="MuiTableFooter-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-footer"
            >
              <td
                class="MuiTableCell-root MuiTableCell-footer MuiTablePagination-root MUIDataTablePagination-root-327"
                colspan="1000"
              >
                <div
                  class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MUIDataTablePagination-toolbar-328 MuiToolbar-gutters"
                >
                  <div
                    class="MuiTablePagination-spacer"
                  />
                  <p
                    class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
                  >
                    عدد الصفوف:
                  </p>
                  <div
                    class="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot MUIDataTablePagination-selectRoot-329"
                  >
                    <div
                      aria-haspopup="listbox"
                      aria-labelledby=" pagination-rows"
                      class="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input MuiInputBase-inputSelect"
                      data-testid="pagination-rows"
                      id="pagination-rows"
                      role="button"
                      tabindex="0"
                    >
                      10
                    </div>
                    <input
                      type="hidden"
                      value="10"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </div>
                  <p
                    class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
                  >
                    1-2 من 2
                  </p>
                  <div
                    class="MuiTablePagination-actions"
                  >
                    <button
                      aria-label="الصفحه السابقه"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                      data-testid="pagination-back"
                      disabled=""
                      id="pagination-back"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                          />
                        </svg>
                      </span>
                    </button>
                    <button
                      aria-label="الصفحه التاليه"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                      data-testid="pagination-next"
                      disabled=""
                      id="pagination-next"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                </div>
              </td>
            </tr>
          </tfoot>
        </table>
        <div
          aria-live="polite"
          class="MUIDataTable-liveAnnounce-149"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
